garage/envs/box2d/box2d_env.py

```diff
@@ -176,11 +176,15 @@ def step(self, action):
         Note: override this method with great care, as it post-processes the
         observations, etc.
         """
-        reward_computer = self.compute_reward(action)
+        # Copy action first to remove side effects
+        action_copy = action if isinstance(action,
```

Comment:
i think you can remove this switch by using the copy stdlib

https://docs.python.org/3.5/library/copy.html